### ISSUE TITLE
[Contribution] qubes-network-server

### ISSUE THREAD

**Issue by andrewdavidwong** [QubesOS/qubes-issues#5088]
**Community Dev:** @Rudd-O
**PoC:** https://github.com/Rudd-O/qubes-network-server
**Discussion:** https://groups.google.com/d/topic/qubes-users/sNjucuuiCcQ/discussion

**Comment by andrewdavidwong** [comment]
@Rudd-O, would you be willing to package your contribution according to our new [package contribution procedure](https://www.qubes-os.org/doc/package-contributions/#contribution-procedure)?

**Comment by Rudd-O** [comment]
I would like to, but it is going to take me some time.  I need to adapt it to Qubes 4 first, then test it real well, then package it according to the procedure.  I'm flagging the e-mail notification so I don't miss it in my queue -- I have to do this work anyway because I use it myself.

**Comment by Rudd-O** [comment]
Yes, but first I have to port it to Qubes 4.  I bought a anew machine and well be assembling it this week, precisely to.do exactly that without messing with my main box.

On June 9, 2019 11:40:28 PM GMT+02:00, Andrew David Wong <notifications@github.com> wrote:
>@Rudd-O, would you be willing to package your contribution according to
>our new [package contribution
>procedure](https://www.qubes-os.org/doc/package-contributions/#contribution-procedure)?
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/QubesOS/qubes-issues/issues/5088#issuecomment-500247785

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by tlaurion** [comment]
@Rudd-O More then a year later, would you like to update the community on what is needed and the current status of the project? 
The Readme https://github.com/Rudd-O/qubes-network-server is a bit misleading.

Thanks!

**Comment by Rudd-O** [comment]
Yes.  It's ported.  Let me take a look at those PRs.

**Comment by Rudd-O** [comment]
I've signed `master` and pushed the signed commit.

I am ready to get this reviewed.  Thanks.

**Comment by andrewdavidwong** [comment]
Assigning to @fepitre for review. Please feel free to reassign as appropriate.

**Comment by tlaurion** [comment]
@fepitre: I see the subject coming again in GSoC. Is this project being reviewed for pointers being given on needed improvements?

**Comment by fepitre** [comment]
@tlaurion I've not found the time for that yet sorry. I've been under a very intensive scheduling since December for multiple Qubes subjects. I'll try to find some soon.

**Comment by DemiMarie** [comment]
I am not super familiar with this project, but I would be more than willing to review any PRs.  @tlaurion would that help?

**Comment by fepitre** [comment]
@DemiMarie at least that would help me to have double checks to approve it.

**Comment by Rudd-O** [comment]
Folks who can review this?

**Comment by fepitre** [comment]
@Rudd-O me this week-end :)

**Comment by Rudd-O** [comment]
I'll be around.

**Comment by fepitre** [comment]
@Rudd-O I would propose several updates, probably in form of PR. Before that, It would be great if you can adjust branches in the following way:

- Dedicated R4.0 branch named as: `release4.0`
- Master branch currently following only R4.1 developments and no specific `r4.1`/`release4.1` as R4.1 is not released.

More generally, we try to make the code common as possible for all releases and we diverge `releaserX.Y` if it's not.

**Comment by fepitre** [comment]
@Rudd-O Here is my review/improvements: https://github.com/fepitre/qubes-network-server/commits/r4.1-improvements. Do you mind to check if there would be any break or such? According to my previous answer, this should be the master branch of the component.

**Comment by fepitre** [comment]
@marmarek I would also vote for renaming this into the service name itself: `qubes-routing-manager`. The current name is too context wide and in my opinion misleading.

**Comment by tlaurion** [comment]
@Rudd-O any updates?

**Comment by Rudd-O** [comment]
Will work on this tomorrow.  I had to travel to Spain due to reasons out of my control.

**Comment by Rudd-O** [comment]
`itshappening.gif`

**Comment by Rudd-O** [comment]
It's complete.  You can deploy with the standard instructions as per the `README.md` file.

**Comment by Rudd-O** [comment]
So, how do we get this upstreamed?

**Comment by Rudd-O** [comment]
I'm thinking my code should mature a bit more and we need to design a UI for it too.

**Comment by fepitre** [comment]
> @Rudd-O: Thank you and congratulations! The next step is for a developer from the Qubes team to review your contribution. For reference, the overall procedure is documented here:
> 
> https://www.qubes-os.org/doc/package-contributions/#contribution-procedure

I already reviewed it few years ago already. I proposed to change the name, and did some split to ensure to have specific content only for dom0 and only for VM, both at once. Any method is fine.

**Comment by Rudd-O** [comment]
Yeah if we meld it within the base OS then we don't have to think about package name.    Alternatively, what features do you want to see integrated, and also what name would be appropriate for this now?
